Model/EventsForThirdPartyModules.php

```diff
@@ -118,19 +135,33 @@ private function prepareForListenerRun($listener)
             return [false, null];
         }
         if (isset($listener["checkClasses"])) {
-            foreach ($listener["checkClasses"] as $className) {
-                if (!$this->doesClassExist($className)) {
+            foreach ($listener["checkClasses"] as $classNameItem) {
+                $classNames = explode('|',$classNameItem);
+                $existClasses = array_filter($classNames, function($className) {
+                    return $this->doesClassExist($className);
+                });
+                if (empty($existClasses)) {
                     return [false,null];
-                }
+                }                
             }
         }
         $sendClasses = [];
         if (isset($listener["sendClasses"])) {
-            foreach ($listener["sendClasses"] as $className) {
-                if (!$this->doesClassExist($className)) {
+            foreach ($listener["sendClasses"] as $classNameItem) {
+                $classNames = explode('|',$classNameItem);
+                $existClasses = array_filter($classNames, function($className) {
+                    return $this->doesClassExist($className);
+                });
+                if (empty($existClasses)) {
+                    return [false,null];
+                }
+                $classInstances = array_map(function($className) {
+                    return $this->objectManager->get($className);
```

Comment:
If two classes exist we create an instance of both of them but use only the first one.
How about stop when we find the first class?

Reply:
I have updated the code, so once the instance is created successfully, stop processing the rest